src/js/utils.js

```diff
@@ -322,6 +322,35 @@ function parseCookie(str, opts) {
   return parsed;
 }
 
+function CheckedDNTBuffer(size) {
+  if (size) {
+    this.size = size;
+  }
+  this.index = 0;
+  this.array = new Array(this.size);
+  this.set = new Set();
+}
+
+CheckedDNTBuffer.prototype = {
+  size: 1000,
+  // if we've seen item, return true. if we haven't store it and return false
+  hasOrSet: function(item) {
```

Comment:
How about `isContaintedAndAddIfNot`?

I do think API misuse is a pretty big concern. I don't trust myself to come back to this code in 2 years and remember that I'm only supposed to call `add` after `has`.